test/cookie.handler.test.js

```diff
@@ -0,0 +1,256 @@
+                                return original.setCookieSync(cookie);
+                            },
+                            getCookieSync: function() {
+                                console.log("ST_LOGS GET_COOKIE_SYNC");
+                                return original.getCookieSync();
+                            }
+                        };
+                    }
+                });
+                let userId = "testing-supertokens-website";
+
+                let loginResponse = await fetch(`${BASE_URL}/login`, {
+                    method: "post",
+                    headers: {
+                        Accept: "application/json",
+                        "Content-Type": "application/json"
+                    },
+                    body: JSON.stringify({ userId })
+                });
+
+                assertEqual(await loginResponse.text(), userId);
+            });
+
+            assert(consoleLogs.includes("ST_LOGS GET_COOKIE"));
+            assert(consoleLogs.includes("ST_LOGS SET_COOKIE sIRTFrontend"));
+            assert(consoleLogs.includes("ST_LOGS SET_COOKIE sAntiCsrf"));
+            assert(consoleLogs.includes("ST_LOGS SET_COOKIE sFrontToken"));
+            // Website SDK does not use the sync functions
+            assert(!consoleLogs.includes("ST_LOGS GET_COOKIE_SYNC"));
+            assert(!consoleLogs.includes("ST_LOGS SET_COOKIE_SYNC sIRTFrontend"));
+            assert(!consoleLogs.includes("ST_LOGS SET_COOKIE_SYNC sAntiCsrf"));
+            assert(!consoleLogs.includes("ST_LOGS SET_COOKIE_SYNC sFrontToken"));
+        } finally {
+            await browser.close();
+        }
+    });
+
+    it("Test that throwing an error in cookie handling gets propogated properly", async function() {
+        await startST();
+        const browser = await puppeteer.launch({
+            args: ["--no-sandbox", "--disable-setuid-sandbox"]
+        });
+
+        try {
+            const page = await browser.newPage();
+
+            page.on("console", event => {
+                const log = event.text();
+
+                console.log(log);
+            });
+
+            await page.goto(BASE_URL + "/index.html", { waitUntil: "load" });
+            await page.addScriptTag({ path: `./bundle/bundle.js`, type: "text/javascript" });
+            await page.evaluate(async () => {
+                let BASE_URL = "http://localhost.org:8080";
+                supertokens.init({
+                    apiDomain: BASE_URL,
+                    cookieHandler: function(original) {
+                        return {
+                            ...original,
+                            getCookie: async function() {
+                                throw new Error("Expected error in tests");
+                            }
+                        };
+                    }
+                });
+                let userId = "testing-supertokens-website";
+                let testFailed = false;
```

Comment:
Ideally we should start with testFailed = true

And then mark it as false only when we get what we expect.